### PR TITLE
[FIX] fix flake8 bugbear errors

### LINKED ISSUE #4378: flake8 bugbear issues
See: https://github.com/PyCQA/flake8-bugbear

```
pip install flake8-bugbear
flake8 --select B nilearn  
```

See the stats below. I would at least tackle B006 and B008 for the reasons explained in those links.

https://docs.astral.sh/ruff/rules/mutable-argument-default/
https://docs.astral.sh/ruff/rules/function-call-in-default-argument/

```
2     B001 Do not use bare `except:`, it also catches unexpected events like memory errors, interrupts, system exit, and so on.  Prefer `except Exception:`.  If you're sure what you're doing, be explicit and write `except BaseException:`.
4     B004 Using `hasattr(x, '__call__')` to test if `x` is callable is unreliable. If `x` implements custom `__getattr__` or its `__call__` is itself not callable, you might get misleading results. Use `callable(x)` for consistent results.
13    B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
13    B007 Loop control variable 'importer' not used within the loop body. If this is intended, start the name with an underscore.
54    B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
2     B009 Do not call getattr with a constant attribute value, it is not any safer than normal property access.
1     B010 Do not call setattr with a constant attribute value, it is not any safer than normal property access.
1     B015 Result of comparison is not used. This line doesn't do anything. Did you intend to prepend it with assert?
2     B017 `assertRaises(Exception)` and `pytest.raises(Exception)` should be considered evil. They can lead to your test passing even if the code being tested is never executed due to a typo. Assert for a more specific exception (builtin or custom), or use `assertRaisesRegex` (if using `assertRaises`), or add the `match` keyword argument (if using `pytest.raises`), or use the context manager form with a target.
2     B018 Found useless Tuple expression. Consider either assigning it to a variable or removing it.
1     B026 Star-arg unpacking after a keyword argument is strongly discouraged, because it only works when the keyword parameter is declared after all parameters supplied by the unpacked sequence, and this change of ordering can surprise and mislead readers.
116   B028 No explicit stacklevel argument found. The warn method from the warnings module uses a stacklevel of 1 by default. This will only show a stack trace for the line on which the warn method is called. It is therefore recommended to use a stacklevel of 2 or greater to provide more information to the user.
```

<details>
<summary>flake8-bugbear (skipped B028 to keep this 'short')</summary>

<pre>
nilearn/_utils/__init__.py:65:13: B007 Loop control variable 'importer' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/_utils/__init__.py:65:32: B007 Loop control variable 'ispkg' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/_utils/data_gen.py:163:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:215:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:270:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:401:26: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:436:9: B007 Loop control variable 'i' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/_utils/data_gen.py:455:26: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:539:50: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:840:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:879:14: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/data_gen.py:882:11: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/_utils/data_gen.py:883:12: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/_utils/niimg_conversions.py:91:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/_utils/tests/test_class_inspect.py:29:46: B026 Star-arg unpacking after a keyword argument is strongly discouraged, because it only works when the keyword parameter is declared after all parameters supplied by the unpacked sequence, and this change of ordering can surprise and mislead readers.
nilearn/_utils/tests/test_masker_validation.py:26:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:218:25: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:233:24: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:255:25: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:255:53: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:269:24: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:269:52: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:292:25: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:292:53: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:300:23: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/conftest.py:300:51: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/connectome/connectivity_matrices.py:432:23: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/connectome/group_sparse_cov.py:554:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/connectome/group_sparse_cov.py:1146:25: B007 Loop control variable 'alpha' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/datasets/atlas.py:732:9: B007 Loop control variable 'n' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/datasets/func.py:1022:17: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/datasets/neurovault.py:2393:22: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/datasets/neurovault.py:2395:17: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/datasets/neurovault.py:2446:22: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/datasets/neurovault.py:2448:17: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/datasets/tests/_testing.py:293:12: B004 Using `hasattr(x, '__call__')` to test if `x` is callable is unreliable. If `x` implements custom `__getattr__` or its `__call__` is itself not callable, you might get misleading results. Use `callable(x)` for consistent results.
nilearn/decoding/decoder.py:970:13: B007 Loop control variable 'i' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/decoding/space_net.py:1266:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/decoding/space_net.py:1498:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/decomposition/_base.py:97:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/decomposition/_base.py:352:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/decomposition/_multi_pca.py:154:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/decomposition/canica.py:173:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/decomposition/dict_learning.py:208:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/decomposition/tests/test_canica.py:21:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/experimental/surface/_maskers.py:54:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/glm/first_level/design_matrix.py:187:16: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/glm/first_level/design_matrix.py:286:16: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/glm/first_level/first_level.py:412:20: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/glm/first_level/first_level.py:418:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/glm/first_level/first_level.py:1040:5: B001 Do not use bare `except:`, it also catches unexpected events like memory errors, interrupts, system exit, and so on.  Prefer `except Exception:`.  If you're sure what you're doing, be explicit and write `except BaseException:`.
nilearn/glm/first_level/first_level.py:1045:9: B001 Do not use bare `except:`, it also catches unexpected events like memory errors, interrupts, system exit, and so on.  Prefer `except Exception:`.  If you're sure what you're doing, be explicit and write `except BaseException:`.
nilearn/glm/first_level/first_level.py:1086:16: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/glm/first_level/first_level.py:1092:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/glm/second_level/second_level.py:414:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/glm/second_level/second_level.py:937:17: B009 Do not call getattr with a constant attribute value, it is not any safer than normal property access.
nilearn/glm/second_level/second_level.py:940:13: B010 Do not call setattr with a constant attribute value, it is not any safer than normal property access.
nilearn/glm/tests/test_first_level.py:1306:32: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/glm/tests/test_utils.py:234:5: B018 Found useless Tuple expression. Consider either assigning it to a variable or removing it.
nilearn/glm/tests/test_utils.py:240:5: B018 Found useless Tuple expression. Consider either assigning it to a variable or removing it.
nilearn/image/image.py:792:13: B015 Result of comparison is not used. This line doesn't do anything. Did you intend to prepend it with assert?
nilearn/image/image.py:1399:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/base_masker.py:25:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/multi_nifti_labels_masker.py:125:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/multi_nifti_maps_masker.py:125:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/multi_nifti_masker.py:150:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/nifti_labels_masker.py:188:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/nifti_maps_masker.py:141:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/nifti_masker.py:66:12: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/nifti_masker.py:258:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/maskers/nifti_spheres_masker.py:304:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/plotting/cm.py:92:12: B004 Using `hasattr(x, '__call__')` to test if `x` is callable is unreliable. If `x` implements custom `__getattr__` or its `__call__` is itself not callable, you might get misleading results. Use `callable(x)` for consistent results.
nilearn/plotting/cm.py:100:26: B007 Loop control variable 'a' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/plotting/cm.py:104:12: B004 Using `hasattr(x, '__call__')` to test if `x` is callable is unreliable. If `x` implements custom `__getattr__` or its `__call__` is itself not callable, you might get misleading results. Use `callable(x)` for consistent results.
nilearn/plotting/cm.py:112:26: B007 Loop control variable 'a' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/plotting/cm.py:327:9: B007 Loop control variable 'c_index' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/plotting/cm.py:355:12: B004 Using `hasattr(x, '__call__')` to test if `x` is callable is unreliable. If `x` implements custom `__getattr__` or its `__call__` is itself not callable, you might get misleading results. Use `callable(x)` for consistent results.
nilearn/plotting/cm.py:361:30: B007 Loop control variable 'a' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/plotting/displays/_slicers.py:1574:13: B007 Loop control variable 'coord' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/plotting/displays/_slicers.py:2064:17: B007 Loop control variable 'coord' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/plotting/html_surface.py:150:63: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/plotting/html_surface.py:194:62: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/plotting/html_surface.py:226:72: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/plotting/surf_plotting.py:1367:34: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/plotting/surf_plotting.py:1368:43: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
nilearn/plotting/tests/test_cm.py:46:5: B017 `assertRaises(Exception)` and `pytest.raises(Exception)` should be considered evil. They can lead to your test passing even if the code being tested is never executed due to a typo. Assert for a more specific exception (builtin or custom), or use `assertRaisesRegex` (if using `assertRaises`), or add the `match` keyword argument (if using `pytest.raises`), or use the context manager form with a target.
nilearn/plotting/tests/test_html_document.py:69:9: B007 Loop control variable 'i' not used within the loop body. If this is intended, start the name with an underscore.
nilearn/plotting/tests/test_html_stat_map.py:26:26: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/plotting/tests/test_img_plotting/test_img_plotting.py:41:38: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/plotting/tests/test_img_plotting/test_plot_anat.py:49:5: B017 `assertRaises(Exception)` and `pytest.raises(Exception)` should be considered evil. They can lead to your test passing even if the code being tested is never executed due to a typo. Assert for a more specific exception (builtin or custom), or use `assertRaisesRegex` (if using `assertRaises`), or add the `match` keyword argument (if using `pytest.raises`), or use the context manager form with a target.
nilearn/regions/parcellations.py:288:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/regions/region_extractor.py:401:16: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/regions/tests/test_signal_extraction.py:57:28: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
nilearn/reporting/glm_reporter.py:492:40: B009 Do not call getattr with a constant attribute value, it is not any safer than normal property access.
nilearn/reporting/html_report.py:269:52: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
</pre>

</details>

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4378

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -12,7 +12,7 @@ exclude =
 	nilearn/_version.py,
 	venv,
 	doc/_build
---select = D,E,F,W,C90,CFQ
+--select = B,D,E,F,W,C90,CFQ
```
Comment (bthirion) [inline .flake8:15]: any hint on what this means ? E.g. A link would be useful

File: nilearn/_utils/data_gen.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def generate_maps(
     border=1,
     window="boxcar",
     random_state=0,
-    affine=np.eye(4),
+    affine=None,
```
Comment (bthirion) [inline nilearn/_utils/data_gen.py:163]: You probably want to update the docstring accordingly

**Review by bthirion — COMMENTED** [review]
The PR is technically good, but the docstrings should be updated accordingly.

**Comment by Remi-Gau** [comment]
> docstrings should be updated accordingly

will do that for the functions that already have doc strings listing parameters.

**Comment by Remi-Gau** [comment]
> I don't understand the doc build failure...

- See this issue : https://github.com/sphinx-doc/sphinx/issues/12300

- This will be fixed according to https://github.com/sphinx-doc/sphinx/issues/12300#issuecomment-2062238457 by the PR https://github.com/nilearn/nilearn/pull/4390

**Pull request merged**